spec/models/admin_dashboard_data_spec.rb

```diff
@@ -3,37 +3,129 @@
+      AdminDashboardData.add_scheduled_problem_check(:test_identifier) do
         called = true
       end
 
-      AdminDashboardData.fetch_problems
-      expect(called).to eq(true)
-
-      AdminDashboardData.fetch_problems(check_force_https: true)
+      AdminDashboardData.execute_scheduled_checks
       expect(called).to eq(true)
     end
 
-    it 'calls the passed method' do
-      $test_AdminDashboardData_global = false
-      class AdminDashboardData
-        def my_test_method
-          $test_AdminDashboardData_global = true
-        end
+    it "adds a found problem from a scheduled check" do
+      AdminDashboardData.add_scheduled_problem_check(:test_identifier) do
+        AdminDashboardData::Problem.maybe_create("test problem")
       end
-      AdminDashboardData.add_problem_check :my_test_method
 
-      AdminDashboardData.fetch_problems
-      expect($test_AdminDashboardData_global).to eq(true)
-      $test_AdminDashboardData_global = nil
+      AdminDashboardData.execute_scheduled_checks
+      problems = AdminDashboardData.load_found_scheduled_check_problems
+      expect(problems.first).to be_a(AdminDashboardData::Problem)
+      expect(problems.first.message).to eq("test problem")
+    end
+
+    it "does not add duplicate problems with the same identifier" do
+      prob1 = AdminDashboardData::Problem.maybe_create("test problem", identifier: "test")
+      prob2 = AdminDashboardData::Problem.maybe_create("test problem 2", identifier: "test")
+      AdminDashboardData.add_found_scheduled_check_problem(prob1)
+      AdminDashboardData.add_found_scheduled_check_problem(prob2)
+      expect(AdminDashboardData.load_found_scheduled_check_problems.count).to eq(1)
+    end
+
+    it "does not error when loading malformed problems saved in redis" do
+      Discourse.redis.set(AdminDashboardData::SCHEDULED_PROBLEM_STORAGE_KEY, "{ 'badjson")
+      expect(AdminDashboardData.load_found_scheduled_check_problems).to eq([])
+    end
+
+    it "clears a specific problem by identifier" do
+      prob1 = AdminDashboardData::Problem.maybe_create("test problem 1", identifier: "test")
+      AdminDashboardData.add_found_scheduled_check_problem(prob1)
+      AdminDashboardData.clear_found_problem("test")
+      expect(AdminDashboardData.load_found_scheduled_check_problems).to eq([])
+    end
+
+    it "defaults to low priority, and uses low priority if an invalid priority is passed" do
+      prob1 = AdminDashboardData::Problem.maybe_create("test problem 1")
+      prob2 = AdminDashboardData::Problem.maybe_create("test problem 2", priority: "superbad")
+      expect(prob1.priority).to eq("low")
+      expect(prob2.priority).to eq("low")
+    end
+  end
+
+  describe 'stats cache' do
+    include_examples 'stats cacheable'
+  end
+
+  describe '#problem_message_check' do
+    let(:key) { AdminDashboardData.problem_messages.first }
+
+    before do
+      described_class.clear_problem_message(key)
+    end
```

Comment:
I think we need to clean up in an `after` block to ensure we don't leak state to other tests.